contributed_definitions/nyaml/NXopt.yaml

```diff
@@ -56,348 +43,675 @@ NXopt(NXobject):
+          doc: |
+            If you specified 'other' as material, decsribe here what it is.
+        thickness(NX_FLOAT):
+          unit: NX_LENGTH
+          doc: |
+            Thickness of the window.
+        orientation_angle(NX_FLOAT):
+          unit: NX_ANGLE
+          doc: |
+            Angle of the window normal (outer) vs. the substrate normal
+            (similar to the angle of incidence).
+      sample_medium_refractive_indices(NX_FLOAT):
+        exists: optional
+        unit: NX_UNITLESS
+        doc: |
+          Array of pairs of complex refractive indices n + ik of the medium
+          for every measured spectral point/wavelength/energy.
+          Only necessary if the measurement was performed not in air, or
+          something very well known, e.g. high purity water.
+        dimensions:
+          rank: 2
+          dim: [[1, 2], [2, N_spectrum]]
+      # beam devices for beam path description
+      (NXbeam_device):
+        exists: [min, 0, max, infty]
+        doc: |
+          Describes the order of respective beam devices in the optical beam
+          path.
+          
+          Everything object which interacts or modifies optical beam properties,
+          may be an beam device, e.g. Filter, Window, Beamsplitter, Photon Source,
+          Detector, etc,
+
+          It is intended, to include this functionality later to "older" beam
+          components, such as NXsource, NXdetector, NXlens, etc.
+          Until this is possbible, auxillary beam devices have to be created,
+          for each "older" beam component instead, to allow a beam path description.
+          To link the auxillary beam device to the real device properties, the
+          attribute \@device should be used.
+        \@device:
+          exists: recommended
+          doc: |
+            Reference to beam device, which is described by a NeXus concept
+            (e.g. for NXsource, entry/instrument/source_TYPE).
+        previous_device:
+          exists: recommended
+          doc: |
+            Reference to the previous beam device, from which the photon beam came
+            to reach this beam device. A photon source is related as origin by ".".
+            This enables a logical order description of the optical setup.
+      # non-beam path elements
+      sample_stage(NXmanipulator):
+        exists: optional
+        doc: |
+          Sample stage (or manipulator) for positioning of the sample. This should
+          only contain the spatial orientation of movement. 
+        stage_type:
+          exists: optional
+          doc: |
+            Specify the type of the sample stage.
+          enumeration: [manual stage, scanning stage, liquid stage, gas cell, cryostat, heater, other]
+        stage_type_other:
+          exists: optional
+          doc: |
+            If "other" was chosen in stage_type, enter here a free text description
+            of the stage type.
+        reference_frame:
```

Comment:
Should be done with `NXtransformations`.

Reply:
yea right, its for now a placeholder. Added it for the todo list